libc/stdio/stdio_file.c

```diff
@@ -665,6 +667,23 @@ TEST(stdio_fileseek, seek_fseek)
 }
 
 
+TEST(stdio_fileseek, seek_fseek_feof)
+{
+	char buf[strlen(teststr) + 1];
+
+	/* fseek does not clear F_EOF flag on error */
+	filep = fopen(STDIO_TEST_FILENAME, "r");
+	TEST_ASSERT_NOT_NULL(filep);
+	{
+		TEST_ASSERT_EQUAL_INT(strlen(teststr), fread(buf, 1, sizeof(buf), filep));
+		TEST_ASSERT_EQUAL_INT(1, feof(filep));
+		TEST_ASSERT_EQUAL_INT(-1, fseek(filep, SEEK_CUR, 10));
+		TEST_ASSERT_EQUAL_INT(1, feof(filep));
```

Comment:
Maybe also check if `F_ERROR` is set after `fseek()` fails with a read or write error? (In this test case or another.)